### ISSUE TITLE
"Avoid serving legacy JavaScript to modern browsers"

### ISSUE THREAD

**Issue by RawMajkel** [ganlanyuan/tiny-slider#685]
__Issue description__:  Google PageSpeed error: _"Avoid serving legacy JavaScript to modern browsers"_ 

__Demo link/slider setting__:  https://imagizer.imageshack.com/v2/767x332q90/922/b5w9fY.jpg 

_Tiny-slider version_:   2.9.3
_Browser name && version_:  not applicable
_OS name && version_:   not applicable

Is it possible to fix this error?

**Comment by dijkermans** [comment]
I don't think that's an error at all, but a minor improvement. imo it's like saying "adding extra code to make it browser compatible is good, but modern browsers don't need it so it may be a waste of extra bytes".

**Comment by RawMajkel** [comment]
@dijkermans Exactly, I don't consider it a bug, but still - would be nice if it gets patched

**Comment by dijkermans** [comment]
Sure, theoretically it'll affect the ranking of a site, but I'm quite sure it's a very minor issue.

**Comment by RawMajkel** [comment]
Yes it does affect, and I'm fed up that my site is 98 and not 100 because of this warning 😂
Of course it's not highest priority but I'd be grately thankful is someone could do a fix of this issue.

**Comment by bobhuf** [comment]
I would like to bump this topic, seeing the Web Vitals penalty has increased.

**Comment by Dredy** [comment]
> I would like to bump this topic, seeing the Web Vitals penalty has increased.

This part, if commented out, removes the error in Lighthouse without removing functionality.

// Object.keys
if (!Object.keys) {
  Object.keys = function(object) {
    var keys = [];
    for (var name in object) {
      if (Object.prototype.hasOwnProperty.call(object, name)) {
        keys.push(name);
      }
    }
    return keys;
  };
}
